### PR TITLE
Add interfaces for Action CheckSnapshotIntegrity

### PR DESCRIPTION
**Pull request by huaxingao**
Co-authored-by: Yufei Gu [yufei@apache.org](yufei@apache.org)
Co-authored-by: Huaxin Gao [huaxin_gao@apple.com](mailto:huaxin_gao@apple.com)

This PR adds  interfaces for Action CheckSnapshotIntegrity. We will have another PR to implement the interfaces in Spark 3.4 and Spark 3.5.

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/actions/CheckSnapshotIntegrity.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package org.apache.iceberg.actions;
+
+import java.util.concurrent.ExecutorService;
+
+public interface CheckSnapshotIntegrity
+    extends Action<CheckSnapshotIntegrity, CheckSnapshotIntegrity.Result> {
+
+  /**
+   * Passes an alternative executor service that will be used for snapshot integrity checking. If
+   * this method is not called, snapshot integrity checker will still be running by a single
```
Comment (himadripal) [inline api/src/main/java/org/apache/iceberg/actions/CheckSnapshotIntegrity.java:28]: nit: 2nd part can be rewritten as  `a single threaded executor service is used by default`

File: api/src/main/java/org/apache/iceberg/actions/CheckSnapshotIntegrity.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+   * Pass the target version to check. The action checks the snapshots in the target version, not in
+   * the current version of the table.
+   *
+   * @param targetVersion the target version file to be checked. Either a file name or a file path
+   *     is acceptable. For example, it could be either
+   *     "00001-8893aa9e-f92e-4443-80e7-cfa42238a654.metadata.json" or
+   *     "/path/to/00001-8893aa9e-f92e-4443-80e7-cfa42238a654.metadata.json".
+   * @return this for method chaining
+   */
+  CheckSnapshotIntegrity targetVersion(String targetVersion);
```
Comment (himadripal) [inline api/src/main/java/org/apache/iceberg/actions/CheckSnapshotIntegrity.java:46]: should the param name be changed to `targetVersionFilePointer`

**Comment by RussellSpitzer** [comment]
I think this is one where we need to have a bit more discussion on the API. I think we've talked previously about how this would be helpful to have as a generic "Repair Snapshot" command which has a couple purposes. 

1) Can check if all data files and manifests in the Snapshot are present
2) Remove missing data files and manifests if requested
3) Rewrite manifests if metadata in manifests doesn't match what is in file footers

Might be good to bring this up on the dev list for discussion

**Comment by szehon-ho** [comment]
Yea looks like there's some discussion in https://github.com/apache/iceberg/pull/10445 and also https://lists.apache.org/thread/6zvpcyvdzglh800v15mt7m7rt76ch3ol.  Initially it makes sense to have a central procedure that can optionally do any of these things.

**Pull request closed without merging**